### PR TITLE
Webc 1450 announcements

### PR DESCRIPTION
**Pull request by ainvoner**

### CODE REVIEW

File: packages/ui/src/announcementManager.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import { AnnouncementContainer } from "./components/announcement-container";
+import { AnnouncementContainerProps } from "./components/announcement-container/announcementContainer";
+
+export interface AnnouncementContent {
+    text: string;
+    title?: string;
+    icon?: any;
+}
+
+export interface AnnouncementProps {
```
Comment (esakal) [inline packages/ui/src/announcementManager.tsx]: rename to `AnnouncementOptions`

File: packages/ui/src/components/announcement-container/announcementContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { Component, h } from "preact";
+import { ContribLogger, getContribLogger } from "@playkit-js-contrib/common";
+import * as styles from "./_announcementContainer.scss";
+
+export interface AnnouncementContainerProps {
+    handleCloseEvent: (event: any) => void;
```
Comment (esakal) [inline packages/ui/src/components/announcement-container/announcementContainer.tsx]: consider rename to `onClose() => void`

File: packages/ui/src/components/announcement-container/announcementContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+export class AnnouncementContainer extends Component<AnnouncementContainerProps> {
+    private _logger: ContribLogger | null = null;
+
+    componentDidMount(): void {
+        this._logger = getContribLogger({
```
Comment (esakal) [inline packages/ui/src/components/announcement-container/announcementContainer.tsx]: move logger to module level

File: packages/ui/src/components/announcement/announcement.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+export interface AnnouncementProps {
+    content: AnnouncementContent;
+}
+
+export class Announcement extends Component<AnnouncementProps> {
+    private _logger: ContribLogger | null = null;
+
+    componentDidMount(): void {
+        this._logger = getContribLogger({
```
Comment (esakal) [inline packages/ui/src/components/announcement/announcement.tsx]: refactor the logger

Comment (ainvoner) [reply packages/ui/src/components/announcement/announcement.tsx]: done

File: packages/ui/src/components/announcement-container/announcementContainer.tsx (hunk 3 above)
Comment (ainvoner) [reply packages/ui/src/components/announcement-container/announcementContainer.tsx]: done

File: packages/ui/src/components/announcement-container/announcementContainer.tsx (hunk 2 above)
Comment (ainvoner) [reply packages/ui/src/components/announcement-container/announcementContainer.tsx]: done

File: packages/ui/src/announcementManager.tsx (hunk 1 above)
Comment (ainvoner) [reply packages/ui/src/announcementManager.tsx]: done

File: packages/ui/src/announcementManager.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+    private _renderAnnouncement(
+        { content, renderContent }: AnnouncementProps,
+        { handleCloseEvent }: AnnouncementContainerProps
+    ) {
+        function _renderContent(overlayItemProps: OverlayItemProps) {
+            return (
+                <AnnouncementContainer handleCloseEvent={handleCloseEvent}>
+                    {renderContent ? (
+                        renderContent(overlayItemProps)
```
[the remark this answers is not in the document]
Comment (ainvoner) [reply packages/ui/src/announcementManager.tsx]: done

File: packages/ui/src/announcementManager.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+            this._options.overlayManager.remove(this._overlayItem);
+            this._overlayItem = null;
+        }
+    }
+
+    reset() {
+        this.remove();
+    }
+
+    private _renderAnnouncement(
```
[the remark this answers is not in the document]
Comment (ainvoner) [reply packages/ui/src/announcementManager.tsx]: done

File: packages/ui/src/announcementManager.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    text: string;
+    title?: string;
+    icon?: any;
+}
+
+export interface AnnouncementProps {
+    content: AnnouncementContent;
+    autoClose?: boolean;
+    duration?: number;
+    renderContent?: (overlayItemProps: OverlayItemProps) => ComponentChild;
```
[the remark this answers is not in the document]
Comment (ainvoner) [reply packages/ui/src/announcementManager.tsx]: done

File: packages/common/src/cuepointEngine.ts [hunk 8]
```diff
@@ -11,63 +11,80 @@ type ChangeData<T extends Cuepoint> = {
...
-	module: 'contrib-common',
-	class: 'CuepointEngine'
+    module: "contrib-common",
+    class: "CuepointEngine"
 });
 
 export class CuepointEngine<T extends Cuepoint> {
-    protected cuepoints: T[];
-
+    private _cuepoints: T[];
```
Comment (esakal) [inline packages/common/src/cuepointEngine.ts]:
```suggestion
    protected _cuepoints: T[];
```

**Pull request merged**